reana-ui/src/pages/launchOnReana/LaunchOnReana.js

```diff
@@ -76,37 +87,107 @@ export default function LaunchOnReana() {
     return !requiredParams.every((param) => qsKeys.includes(param));
   }
 
+  /**
+   * Memoised function to get workflow parameters.
+   * Since this function is called twice and its return value should not
+   * change unless the query string parameters change, memoising it improves
+   * performace, avoiding extra calls.
+   */
+  const getWorkflowParameters = useCallback(() => {
+    /**
+     * Parses workflow parameters. Triggers notification error if invalid.
+     * @param {URLSearchParams} query Standard URLSearchParams object.
+     * @returns {Object | null} Returns object with workflow parameters or `null` if invalid.
+     */
+    function fn(query) {
+      try {
+        return JSON.parse(query.get("parameters"));
+      } catch (error) {
+        dispatch(
+          triggerNotification(
+            "An error has occurred",
+            `Invalid JSON workflow parameters provided: ${query.get(
+              "parameters"
+            )}`,
+            { error: true }
+          )
+        );
+      }
+      return null;
+    }
+    return fn(query);
+  }, [dispatch, query]);
+
   return (
     <BasePage>
       <Container text className={styles["container"]}>
         <div>
           <Title>Launch on REANA</Title>
         </div>
         {isMissingRequiredParams(query) ? (
-          <Message
-            icon="warning circle"
-            header="Missing required query string fields."
-            content={`The required fields: ${getRequiredQsParams()}`}
-            warning
-          />
+          <Welcome />
         ) : (
-          <ul>
-            {[...query].map(([k, v]) => (
-              <li key={k}>
-                {k}: {v}
-              </li>
-            ))}
-          </ul>
+          <Box className={styles.box} wrap>
+            <section className={styles.details}>
+              <span className={styles.icon}>
+                <Icon name="hourglass half" />
+              </span>
+              <div>
+                <div className={styles.name}>
+                  {query.get("name") ?? DEFAULT_WORKFLOW_NAME}
+                </div>
+                <a className={styles.url} href={query.get("url")}>
+                  {query.get("url")}
+                </a>
+                <div>
+                  <Icon name="file code outline" />
+                  {query.get("spec") ?? DEFAULT_SPEC_FILENAME}
+                </div>
+                {query.get("parameters") && getWorkflowParameters() && (
+                  <div className={styles.parameters}>
+                    Parameters:
+                    <ul>
+                      {Object.entries(getWorkflowParameters()).map(([k, v]) => (
```

Comment:
Good observation. I had an attempt to use `useCallback()` when declaring the function to prevent the function to be called twice, but after checking it again it seems like it's not doing the job properly. I'll replace it.